test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -1102,10 +1101,16 @@ private Object runExtensionMethod(ReflectiveInvocationContext<Method> invocation
 
     private Method determineTCCLExtensionMethod(Method originalMethod, Class<?> c)
             throws ClassNotFoundException {
-        Method newMethod = tcclMethodCache.get(originalMethod);
-        if (newMethod != null) {
-            return newMethod;
+
+        // cacheKey contains identity hash code of the method instead of the more expensive toString
+        // and it has to be "contextualized" with the given class (here, name is good enough)
+        final String cacheKey = System.identityHashCode(originalMethod) + "_" + c.getName();
```

Comment:
Hash codes are not really guaranteed to be unique, so there is the possibility here for this to go wrong.

Reply:
Are you sure the _identity_ hash code can clash? This is _not_ the regular hash code of `method` that doesn't contain method parameters (which can clash easily).

Reply:
Are we sure this cache brings something to the plate? Because I see a lot of questions and adjustments and I haven't seen any proof this was any useful?

Reply:
It does cut down the invocation quantity quite a bit: https://github.com/quarkusio/quarkus/pull/19751#discussion_r698059833
But I haven't done any runtime measurements, so call it premature optimization if you like.
Let me check real quick...

Reply:
I decided to remove the cache as the benefit is barely measurable and there seems to be no "efficient" cache key that covers all concerns.